tests/python/frontend/test_common.py

```diff
@@ -27,6 +32,203 @@ def test_key_is_not_present():
+
+            c_2 = set_span(relay.const(np.zeros(10)), "zeros")
+            y = set_span(relay.add(body, c_2), "add_2")
+            return relay.Function([x], y)
+
+        def _golden():
+            x = relay.Var("x", span=_create_span("x_var"))
+            c_1 = relay.const(np.ones(10), span=_create_span("let"))
+            add = _set_span(relay.add(x, x), "let")
+            body = relay.Let(x, c_1, add, span=_create_span("let"))
+
+            c_2 = relay.const(np.zeros(10), span=_create_span("zeros"))
+            y = _set_span(relay.add(body, c_2), "add_2")
+            return relay.Function([x], y)
+
+        assert _verify_structural_equal_with_span(_res(), _golden())
+
+    def _verify_if():
+        def _res():
+            x = set_span(relay.var("x", shape=[], dtype="float32"), "x_var")
+            y = set_span(relay.var("y", shape=[], dtype="float32"), "y_var")
+            eq = relay.equal(x, y)
+
+            true_branch = set_span(relay.add(x, y), "true_branch")
+            false_branch = relay.subtract(x, y)
+            ife = set_span(relay.If(eq, true_branch, false_branch), "if")
+            return relay.Function([x, y], ife)
+
+        def _golden():
+            x = relay.var("x", shape=[], dtype="float32", span=_create_span("x_var"))
+            y = relay.var("y", shape=[], dtype="float32", span=_create_span("y_var"))
+            eq = _set_span(relay.equal(x, y), "if")
+
+            true_branch = _set_span(relay.add(x, y), "true_branch")
+            false_branch = _set_span(relay.subtract(x, y), "if")
+            ife = relay.If(eq, true_branch, false_branch, span=_create_span("if"))
+            return relay.Function([x, y], ife)
+
+        assert _verify_structural_equal_with_span(_res(), _golden())
+
+    def _verify_fn():
+        def _res():
+            x = set_span(relay.var("x", shape=(1, 64, 56, 56)), "x_var")
+            w = relay.const(np.ones([64, 64, 3, 3]), dtype="int64")
+            y = relay.nn.conv2d(x, w, channels=64, kernel_size=(3, 3), padding=(1, 1))
+            f = set_span(relay.Function([x], y), "func")
+            return f
+
+        def _golden():
+            x = relay.var("x", shape=(1, 64, 56, 56), span=_create_span("x_var"))
+            w = relay.const(np.ones([64, 64, 3, 3]), dtype="int64", span=_create_span("func"))
+            y = _set_span(
+                relay.nn.conv2d(x, w, channels=64, kernel_size=(3, 3), padding=(1, 1)), "func"
+            )
+            f = relay.Function([x], y, span=_create_span("func"))
+            return f
+
+        assert _verify_structural_equal_with_span(_res(), _golden())
+
+    _verify_env_var_switch()
+    _verify_builtin_tuple()
+    _verify_builtin_list()
+    _verify_var()
+    _verify_constant()
+    _verify_call()
+    _verify_tuple()
+    _verify_tuple_getitem()
+    _verify_let()
+    _verify_if()
+    _verify_fn()
+
+
 if __name__ == "__main__":
     test_key_is_present()
```

Comment:
can you change this to `tvm.testing.main()` so the test cases are split out ?

Reply:
Will do.